Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -797,22 +829,33 @@ lemma map_polynomial : (W.map f).toAffine.polynomial = W.polynomial.map (mapRing
   simp only [polynomial]
   map_simp
 
-lemma map_eval_polynomial (x : R[X]) (y : R) :
+lemma map_polynomial_eval (x : R[X]) :
+    (W.map f).toAffine.polynomial.eval (x.map f) = (W.polynomial.eval x).map f := by
+  rw [map_polynomial, eval_map, ← coe_mapRingHom, eval₂_hom]
+
+lemma map_polynomial_eval_eval (x : R[X]) (y : R) :
     ((W.map f).toAffine.polynomial.eval <| x.map f).eval (f y) =
       f ((W.polynomial.eval x).eval y) := by
-  rw [map_polynomial, eval_map, ← coe_mapRingHom, eval₂_hom, coe_mapRingHom, eval_map, eval₂_hom]
+  rw [map_polynomial_eval, eval_map, eval₂_hom]
+
+lemma map_polynomial_evalEval (x y : R) :
+    (W.map f).toAffine.polynomial.evalEval (f x) (f y) = f (W.polynomial.evalEval x y) := by
+  rw [← map_polynomial_eval_eval, map_C]
+
+lemma Equation.map {x y : R} (h : W.Equation x y) : Equation (W.map f) (f x) (f y) := by
```

Comment:
Urgh, I just noticed that some of my `map` lemmas might not have the correct name: `map_*` should be a lemma of the form `f (...) = (f ...)`, but some of mine change the Weierstrass curve instead? I don't know what's the best naming convention, but maybe you can think of something that's analogous to the distinction between `Equation.map` and `map_polynomial_*`. Surely you should add the analogue of all of the new additions for `Nonsingular` too...

Reply:
As I remarked [in the Jacobian lemmas PR](https://github.com/leanprover-community/mathlib4/pull/13846#discussion_r1641567888), the other naming convention is also used, e.g. in [Prod.map_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Prod/Basic.html#Prod.map_injective) and many more `map_injective` lemmas.

What do you want me to add about `Nonsingular`? `Nonsingular.map` isn't necessarily true for non-injective RingHom.

Reply:
Fair enough!